### PR TITLE
Update documentation to make the project more discoverable

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Various updates to the README and other documentation pages to reflect the latest state of the projects and make the project more discoverable.

Here's the changes and reasoning:

#### CHANGELOG.md
- Remove the empty future section as just confusing and no longer required since we get these from GitHub draft release since #1295 
- Made the comment an HTML comment so it doesn't show for anyone viewing the changes outside of source.

#### CONTRIBUTING.md
- Updated example to 0.6.0 as 0.4.0 was a bit dated now

#### README.md
- Correct heading level and capitalisation for our headline ("The SQL Linter for Humans")
- Update GitHub badges broken since we moved from `master` to `main`.
- Highlighted SQLFLuff in bold consistently.
- Added section on Dialects supported and Templates supported. This will require some manual maintenance admittedly but think it's better for searchability and to let potential users easily see what we support (and how many different dialects we support so adding new ones is entirely possible!)
- Add hyperlinks to CLI and Rules for ease of use.
- Added note that the docs website is maintained from this repo so changes can be suggested here.
- Hide long Slack URL and made it a link instead.
- Added Twitter link.
- Reformatted Contributing section line endings and added link to GitHub contributors.

#### docs.sqlfluff.com
- Consistent capitalisation for our tagline ("The SQL Linter for Humans")
- Added Twitter link

#### setup.py (for PyPi)
- Consistent description ("The SQL Linter for Humans")
- Removed embedded CHANGELOG and added link instead
- Added more keywords for dialects we support + dbt. I didn't bother adding Jinja as that's quite generic.
- Added more links (based on [Flask](https://pypi.org/project/Flask/) which has these link types).

#### GitHub Issue Templates
- Adds a Documentation issue template for future suggested changes.

### Are there any other side effects of this change that we should be aware of?

Shouldn't be!

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/parser` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: docs/source/jointhecommunity.rst [hunk 1]
```diff
@@ -6,3 +6,11 @@ SQLFluff Slack
 We have a fast-growing community on Slack, come and join us!
 
 https://join.slack.com/t/sqlfluff/shared_invite/zt-o1f4x0e8-pZzarAIlQmKj_6ZwD16w0g
```
Comment (zhongjiajie) [inline docs/source/jointhecommunity.rst:8]:
```suggestion
We have a fast-growing community [on Slack](https://join.slack.com/t/sqlfluff/shared_invite/zt-o1f4x0e8-pZzarAIlQmKj_6ZwD16w0g), come and join us!
```

File: README.md [hunk 2]
```diff
@@ -31,13 +56,13 @@ L:   1 | P:  14 | L006 | Operators should be surrounded by a single space unless
...
-You can also have a play using [SQLFluff online](https://online.sqlfluff.com/).
+You can also have a play using [**SQLFluff online**](https://online.sqlfluff.com/).
 
-For full CLI usage and rules reference, see the docs.
+For full [CLI usage](https://docs.sqlfluff.com/en/stable/cli.html) and [rules reference](https://docs.sqlfluff.com/en/stable/rules.html), see [the SQLFLuff docs](https://docs.sqlfluff.com/en/stable/).
 
 # Documentation
 
-For full documentation visit [docs.sqlfluff.com](https://docs.sqlfluff.com/en/stable/).
+For full documentation visit [docs.sqlfluff.com](https://docs.sqlfluff.com/en/stable/). This documentation is generated from this repository so please raise [issues](https://github.com/sqlfluff/sqlfluff/issues) or pull requests for any additions, corrections, or clarifications.
```
Comment (zhongjiajie) [inline README.md:65]:
```suggestion
For full documentation visit [docs.sqlfluff.com](https://docs.sqlfluff.com/en/stable). This documentation is generated from this repository so please raise [issues](https://github.com/sqlfluff/sqlfluff/issues) or pull requests for any additions, corrections, or clarifications.
```

File: README.md [hunk 3]
```diff
@@ -31,13 +56,13 @@ L:   1 | P:  14 | L006 | Operators should be surrounded by a single space unless
 L:   1 | P:  27 | L001 | Unnecessary trailing whitespace
 ```
 
-You can also have a play using [SQLFluff online](https://online.sqlfluff.com/).
+You can also have a play using [**SQLFluff online**](https://online.sqlfluff.com/).
 
-For full CLI usage and rules reference, see the docs.
+For full [CLI usage](https://docs.sqlfluff.com/en/stable/cli.html) and [rules reference](https://docs.sqlfluff.com/en/stable/rules.html), see [the SQLFLuff docs](https://docs.sqlfluff.com/en/stable/).
```
Comment (zhongjiajie) [inline README.md:61]:
```suggestion
For full [CLI usage](https://docs.sqlfluff.com/en/stable/cli.html) and [rules reference](https://docs.sqlfluff.com/en/stable/rules.html), see [the SQLFLuff docs](https://docs.sqlfluff.com/en/stable).
```

**Review by zhongjiajie — COMMENTED** [review]
LGTM, and I add some minor suggest.

File: setup.py [hunk 4]
```diff
@@ -33,19 +33,35 @@ def read(*names, **kwargs):
...
+        "snowflake",
+        "teradata",
+        "dbt",
+    ],
     project_urls={
         "Homepage": "https://www.sqlfluff.com",
         "Documentation": "https://docs.sqlfluff.com",
+        "Changes": "https://github.com/sqlfluff/sqlfluff/blob/main/CHANGELOG.md",
         "Source": "https://github.com/sqlfluff/sqlfluff",
+        "Issue Tracker": "https://github.com/sqlfluff/sqlfluff/issues/",
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply setup.py]: Weird that GitHub doesn't enforce one over the other. But agree, let's us the one they use in their links.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ![SQLFluff](https://raw.githubusercontent.com/sqlfluff/sqlfluff/master/images/sqlfluff-wide.png)
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply README.md]: Good spot!

File: setup.py [hunk 6]
```diff
@@ -33,19 +33,35 @@ def read(*names, **kwargs):
...
     author="Alan Cruickshank",
     author_email="alan@designingoverload.com",
     url="https://github.com/sqlfluff/sqlfluff",
     python_requires=">=3.6",
-    keywords=["sqlfluff", "sql", "linter"],
+    keywords=[
+        "sqlfluff",
+        "sql",
+        "linter",
+        "formatter",
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply setup.py:48]: Bit subjective here but my gut feel is no. Although the parse is a big part of what SQLFluff is, it's not really what it does but more how it does it. In general it's a linter and auto fixer for various languages. The fact it does this by lexing and parsing is what makes it good, but not really a keyword IMHO. If people are searching for a "parser" we wouldn't want them to find us. Whereas people searching for "formatter" or "linter" or "SQL" we would want them to find us.

File: docs/source/jointhecommunity.rst (hunk 1 above)
Comment (tunetheweb) [reply docs/source/jointhecommunity.rst:8]: Yeah could inline this link - consistent with the README. However it's an .rst file and not markdown to syntax is different. Will add separately.

File: README.md (hunk 2 above)
Comment (tunetheweb) [reply README.md:65]: I disagree with this suggestion. Visiting https://docs.sqlfluff.com/en/stable explicitly redirects to the trailing slash version: https://docs.sqlfluff.com/en/stable/ so let's save users the effort of the redirect.

File: README.md (hunk 3 above)
Comment (tunetheweb) [reply README.md:61]: Same comment here. We use trailing slashes on the docs website.

File: docs/source/jointhecommunity.rst [hunk 7]
```diff
@@ -6,3 +6,11 @@ SQLFluff Slack
...
+
+
+SQLFluff on Twitter
+====================
+
+Follow us On Twitter `@SQLFLuff <https://twitter.com/SQLFluff>`_ for annoncements and other related posts.
+
+https://twitter.com/SQLFluff
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply docs/source/jointhecommunity.rst:16]: Fair point! will remove that and the new line above it.

**Comment by tunetheweb** [comment]
Thanks @zhongjiajie for the review!

**Review by NiallRees — APPROVED** [review]

**Pull request merged**